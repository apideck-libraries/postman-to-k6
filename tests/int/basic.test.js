/* eslint-disable space-before-function-paren */
import convertFile from '../../lib/convert/file';

test('minimal', async () => {
  const [main] = await convertFile('test/material/2/minimal.json');
  expect(main).toBe('// No HTTP/HTTPS transactions have been recorded');
});

test('request', async () => {
  const [main] = await convertFile('test/material/2/request.json');
  expect(main).toMatchSnapshot();
});

test('raw body', async () => {
  const [main] = await convertFile('test/material/2/body-raw.json');
  expect(main).toMatchSnapshot();
});

test('form body text', async () => {
  const [main] = await convertFile('test/material/2/body-form.json');
  expect(main).toMatchSnapshot();
});

test('form body file', async () => {
  const [main] = await convertFile('test/material/2.1/upload.json');
  expect(main).toMatchSnapshot();
});

test('url body', async () => {
  const [main] = await convertFile('test/material/2/body-url.json');
  expect(main).toMatchSnapshot();
});

test('no body alternate', async () => {
  const [main] = await convertFile('test/material/2.1/no-body-alternate.json');
  expect(main).toMatchSnapshot();
});

test('iterations', async () => {
  const [main] = await convertFile('test/material/2/request.json', {
    iterations: 25,
  });
  expect(main).toMatchSnapshot();
});
