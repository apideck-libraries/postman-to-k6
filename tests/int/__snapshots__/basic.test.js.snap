// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`form body file 1`] = `
"// Auto-generated by the postman-to-k6 converter

import "./libs/shim/core.js";
import http from "k6/http";

export let options = { maxRedirects: 4 };

const Request = Symbol.for("request");
postman[Symbol.for("initial")]({
  options
});

const files = {};
files["theultimatequestion.txt"] = http.file(
  open("theultimatequestion.txt", "b"),
  "theultimatequestion.txt"
);

export default function() {
  postman[Request]({
    name: "TestFileUpload",
    method: "POST",
    address: "http://deepthought.test/",
    data: {
      file: files["theultimatequestion.txt"]
    }
  });
}
"
`;

exports[`form body text 1`] = `
"// Auto-generated by the postman-to-k6 converter

import "./libs/shim/core.js";

export let options = { maxRedirects: 4 };

const Request = Symbol.for("request");
postman[Symbol.for("initial")]({
  options
});

export default function() {
  postman[Request]({
    name: "TestRequest",
    method: "POST",
    address: "http://example.com/",
    data: {
      first: "one",
      second: "two",
      third: "three"
    }
  });
}
"
`;

exports[`iterations 1`] = `
"// Auto-generated by the postman-to-k6 converter

import "./libs/shim/core.js";

export let options = { maxRedirects: 4, iterations: 25 };

const Request = Symbol.for("request");
postman[Symbol.for("initial")]({
  options
});

export default function() {
  postman[Request]({
    name: "TestRequest",
    method: "GET",
    address: "http://example.com/"
  });
}
"
`;

exports[`no body alternate 1`] = `
"// Auto-generated by the postman-to-k6 converter

import "./libs/shim/core.js";

export let options = { maxRedirects: 4 };

const Request = Symbol.for("request");
postman[Symbol.for("initial")]({
  options
});

export default function() {
  postman[Request]({
    name: "TestRequest",
    method: "GET",
    address: "http://example.com/"
  });
}
"
`;

exports[`raw body 1`] = `
"// Auto-generated by the postman-to-k6 converter

import "./libs/shim/core.js";

export let options = { maxRedirects: 4 };

const Request = Symbol.for("request");
postman[Symbol.for("initial")]({
  options
});

export default function() {
  postman[Request]({
    name: "TestRequest",
    method: "POST",
    address: "http://example.com/",
    data: "line1\\nline2\\nline3\\n"
  });
}
"
`;

exports[`request 1`] = `
"// Auto-generated by the postman-to-k6 converter

import "./libs/shim/core.js";

export let options = { maxRedirects: 4 };

const Request = Symbol.for("request");
postman[Symbol.for("initial")]({
  options
});

export default function() {
  postman[Request]({
    name: "TestRequest",
    method: "GET",
    address: "http://example.com/"
  });
}
"
`;

exports[`url body 1`] = `
"// Auto-generated by the postman-to-k6 converter

import "./libs/shim/core.js";

export let options = { maxRedirects: 4 };

const Request = Symbol.for("request");
postman[Symbol.for("initial")]({
  options
});

export default function() {
  postman[Request]({
    name: "TestRequest",
    method: "POST",
    address: "http://example.com/",
    data: {
      first: "one",
      second: "two",
      third: "three"
    },
    headers: {
      "Content-Type": "application/x-www-form-urlencoded"
    }
  });
}
"
`;
